### DISCUSSION TITLE
Latest update libmesh compile failure

### DISCUSSION THREAD

**Discussion by srinath-chakravarthy** [idaholab/moose#17203] [Q&A Getting Started] [answered]
I updated moose this morning and get the following error on both master and next branches 

/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C: In instantiation of ‘void libMesh::PetscLinearSolver<T>::get_residual_history(std::vector<double>&) [with T = double]’:
/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:1562:16:   required from here
/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:1274:39: error: invalid conversion from ‘const PetscReal**’ {aka ‘const double**’} to ‘PetscReal**’ {aka ‘double**’} [-fpermissive]
 1274 |   ierr = KSPGetResidualHistory(_ksp, &p, &its);
      |                                       ^
      |                                       |
      |                                       const PetscReal** {aka const double**}
In file included from /home/srinath/github/petsc/include/petscsnes.h:6,
                 from /home/srinath/github/petsc/include/petscts.h:7,
                 from /home/srinath/github/petsc/include/petsc.h:30,
                 from ./include/libmesh/petsc_macro.h:66,
                 from ./include/libmesh/petsc_linear_solver.h:27,
                 from /home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:26:
/home/srinath/github/petsc/include/petscksp.h:140:55: note:   initializing argument 2 of ‘PetscErrorCode KSPGetResidualHistory(KSP, PetscReal**, PetscInt*)’
  140 | PETSC_EXTERN PetscErrorCode KSPGetResidualHistory(KSP,PetscReal*[],PetscInt*);
      |                                                       ^~~~~~~~~~~~
/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C: In instantiation of ‘libMesh::Real libMesh::PetscLinearSolver<T>::get_initial_residual() [with T = double; libMesh::Real = double]’:
/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:1562:16:   required from here
/home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:1316:39: error: invalid conversion from ‘const PetscReal**’ {aka ‘const double**’} to ‘PetscReal**’ {aka ‘double**’} [-fpermissive]
 1316 |   ierr = KSPGetResidualHistory(_ksp, &p, &its);
      |                                       ^
      |                                       |
      |                                       const PetscReal** {aka const double**}
In file included from /home/srinath/github/petsc/include/petscsnes.h:6,
                 from /home/srinath/github/petsc/include/petscts.h:7,
                 from /home/srinath/github/petsc/include/petsc.h:30,
                 from ./include/libmesh/petsc_macro.h:66,
                 from ./include/libmesh/petsc_linear_solver.h:27,
                 from /home/srinath/github/moose/scripts/../libmesh/src/solvers/petsc_linear_solver.C:26:
/home/srinath/github/petsc/include/petscksp.h:140:55: note:   initializing argument 2 of ‘PetscErrorCode KSPGetResidualHistory(KSP, PetscReal**, PetscInt*)’

**Comment by GiudGiud** [comment, marked as answer]
Hi

Does it work now?
DId you get the libmesh update through conda update --all ?

Guillaume
